pkg/extflag/pathorcontent.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+// Taken from Thanos project.
+//
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+package extflag
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+// PathOrContent is a flag type that defines two flags to fetch bytes. Either from file (*-file flag) or content (* flag).
+type PathOrContent struct {
+	flagName string
+
+	envSubstitution bool
+	required        bool
+
+	path    *string
+	content *string
+}
+
+type FlagClause interface {
+	Flag(name, help string) *kingpin.FlagClause
+}
+
+// RegisterPathOrContent registers PathOrContent flag in kingpinCmdClause.
+func RegisterPathOrContent(cmd FlagClause, flagName string, help string, required bool, envSubstitution bool) *PathOrContent {
+	fileFlagName := fmt.Sprintf("%s-file", flagName)
+	contentFlagName := flagName
+
+	fileHelp := fmt.Sprintf("Path to %s", help)
+	fileFlag := cmd.Flag(fileFlagName, fileHelp).PlaceHolder("<file-path>").String()
+
+	contentHelp := fmt.Sprintf("Alternative to '%s' flag (mutually exclusive). Content of %s", fileFlagName, help)
+	contentFlag := cmd.Flag(contentFlagName, contentHelp).PlaceHolder("<content>").String()
+
+	return &PathOrContent{
+		flagName:        flagName,
+		required:        required,
+		path:            fileFlag,
+		content:         contentFlag,
+		envSubstitution: envSubstitution,
+	}
+}
+
+// Content returns the content of the file when given or directly the content that has been passed to the flag.
+// It returns an error when:
+// * The file and content flags are both not empty.
+// * The file flag is not empty but the file can't be read.
+// * The content is empty and the flag has been defined as required.
+func (p *PathOrContent) Content() ([]byte, error) {
+	fileFlagName := fmt.Sprintf("%s-file", p.flagName)
+
+	if len(*p.path) > 0 && len(*p.content) > 0 {
+		return nil, errors.Errorf("both %s and %s flags set.", fileFlagName, p.flagName)
+	}
+
+	var content []byte
+	if len(*p.path) > 0 {
+		c, err := ioutil.ReadFile(*p.path)
+		if err != nil {
+			return nil, errors.Wrapf(err, "loading file %s for %s", *p.path, fileFlagName)
+		}
+		content = c
+	} else {
+		content = []byte(*p.content)
+	}
+
+	if len(content) == 0 && p.required {
+		return nil, errors.Errorf("flag %s or %s is required for running this command and content cannot be empty.", fileFlagName, p.flagName)
+	}
+	if p.envSubstitution {
+		content = substituteEnvVars(string(content))
+	}
+	return content, nil
+}
+
+// substituteEnvVars returns content of YAML file with substituted environment variables.
+// Will be substituted with empty string if env var isn't set.
+// Follows K8s convention, i.e $(...), as mentioned here https://kubernetes.io/docs/tasks/inject-data-application/define-interdependent-environment-variables/.
+func substituteEnvVars(content string) []byte {
```

Comment:
Let's take in bytes if we want to split out bytes (consistency)

Reply:
Can we reuse exsiting code? https://github.com/thanos-io/thanos/blob/7d118f416be98cf502b9b5e0df4e98b37478f3f3/pkg/reloader/reloader.go#L419

Reply:
Alright! Will try to reuse code from here!

Reply:
Have taken existing code but used `os.Getenv` instead of `os.LookupEnv` since `Token` here is optional and don't want to error out if variable is unset. 🙂

Reply:
Why we don't want to error out? I think that is a correct behaviour

Reply:
The `GITHUB_TOKEN` variable is only expected to be set in CI runners so that authenticated API calls are made and it's not rate limited(Same IP across macos runners problem). But if user decides not to set the token i.e, isn't getting rate limited and doesn't need to make auth'd requests then there shouldn't be any error I think. So should I put the error back in?

Reply:
>  But if user decides not to set the token i.e, isn't getting rate limited and doesn't need to make auth'd requests then there shouldn't be any error I think. 

If user decided not to set this token, why adding this to yaml as `$(....)` construct?

Reply:
If they wanted to run CI job locally but then they can have a make-docs-local command as well. I see what you mean, I'll add error back in! 🙂

Reply:
explicit changes are better than surprising ones